src/sentry/ingest/userreport.py

```diff
@@ -93,10 +107,119 @@ def find_event_user(report_data, event):
             return None
 
     tag = event.get_tag("sentry:user")
+
     if not tag:
         return None
 
     try:
         return EventUser.for_tags(project_id=report_data["project_id"], values=[tag])[tag]
     except KeyError:
         pass
+
+
+def find_event_user_with_snuba(event: Event, eventuser: EventUser):
+    """
+    Query Snuba to get the EventUser information for an Event.
+    Then compare with the EventUser record and confirm that the
+    query result is equivalent.
+    """
+    start_date, end_date = _start_and_end_dates(event.datetime)
+
+    query = _generate_entity_dataset_query(
+        event.project_id, event.group_id, event.event_id, start_date, end_date
+    )
+    request = Request(
+        dataset=Dataset.Events.value,
+        app_id=REFERRER,
+        query=query,
+        tenant_ids={"referrer": REFERRER, "organization_id": event.project.organization.id},
+    )
+    data_results = raw_snql_query(request, referrer=REFERRER)["data"]
+
+    if len(data_results) == 0:
+        return logger.info(
+            "Errors dataset query to find EventUser did not return any results.",
+            extra={
+                "eventuser": serialize(eventuser),
+                "event_id": event.event_id,
+                "project_id": event.project_id,
+                "group_id": event.group_id,
+            },
+        )
+
+    if not _is_equal_to_eventuser(eventuser, data_results[0]):
```

Comment:
There's actually a reasonable chance that these won't match up. Once an `EventUser` row is written, then it never changes. But the event could have more up to date `EventUser` data.

We determine the id of an `EventUser` based on a hierarchy of fields. Someting like `ident`, `email`, `ip_address`. Whichever field is present first is used as the id for the `EventUser`.

So it's possible that an event might have `ident=123`, which would then be used as the id for the `EventUser` row. But then the event might have an email like `dan@sentry.io`, and the `EventUser` might have `nisanthan@sentry.io`.

IMO this isn't a problem at all, but if you do see discrepancies that's one potential reason why.